src/main/java/org/jenkinsci/plugins/codesonar/services/CodeSonarCacheService.java

```diff
@@ -30,14 +32,26 @@ public class CodeSonarCacheService {
     private MetricsService metricsService;
     private ProceduresService proceduresService;
     private AlertsService alertsService;
+    private WarningsService warningsService;
     private Map<Pair<URI,Long>, CodeSonarAnalysisData> cachedAnalyses;
 
-    public CodeSonarCacheService(HttpService httpService, CodeSonarHubInfo hubInfo) {
+    private CodeSonarCacheService(HttpService httpService, CodeSonarHubInfo hubInfo) {
         this.httpService = httpService;
         this.hubInfo = hubInfo;
         cachedAnalyses = new HashMap<>();
     }
     
+    public static CodeSonarCacheService createInstance(HttpService httpService, CodeSonarHubInfo hubInfo) {

```

Comment:
This is the method that handles the creation of a new instance (first time only) as well as the one that can return that same instance when already available.